app/Http/Controllers/AdminConabController.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\User;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Auth;
+use Illuminate\Support\Facades\Validator;
+
+class AdminConabController extends Controller
+{
+    /**
+     * Display a listing of the resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function index()
+    {
+        $user = Auth::user();
+        $users = User::where('user_type', 'ADMIN_CONAB')->where('id', '<>', $user->id)->get();
+        return $users;
+    }
+
+    /**
+     * Store a newly created resource in storage.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @return \Illuminate\Http\Response
+     */
+    public function store(Request $request)
+    {
+        $data = Validator::make($request->all(), [
+            'name' => 'required|string',
+            'email' => 'required|email',
+            'cpf' => 'required|regex:/^[0-9]{3}\.[0-9]{3}\.[0-9]{3}\-[0-9]{2}/',
+            'phones' => 'required',
+            'phones.*.number' => 'required|string|regex:/^\([0-9]{2}\) [0-9]{5}\-[0-9]{4}/'
+        ])->validate();
+
+        $user = new User();
+        $user->fill($data);
+        $user->password = $data['cpf'];
+        $user->user_type = 'ADMIN_CONAB';
+        $user->save();
+        $phones = $user->phones()->createMany($data['phones']);
+        $userAndPhones = array_merge($user->toArray(), [ 'phones' => $phones ]);
+
+        return response($userAndPhones, 201);
+    }
+
+    /**
+     * Update the specified resource in storage.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @param  int  $id
+     * @return \Illuminate\Http\Response
+     */
+    public function update(Request $request, $id)
+    {
+        $data = Validator::make($request->all(), [
+            'name' => 'string',
+            'email' => 'string|email',
+            'cpf' => 'regex:/^[0-9]{3}\.[0-9]{3}\.[0-9]{3}\-[0-9]{2}/',
+            'password' => 'string',
+            'new_password' => 'string',
+            'phones.*.number' => 'string|regex:/^\([0-9]{2}\) [0-9]{5}\-[0-9]{4}/'
+        ])->validate();
+
+        $admin = User::with('phones')->findOrFail($id);
+        $admin->name = $data['name'] ?? $admin->name;
+        $admin->email = $data['email'] ?? $admin->email;
+        $admin->cpf = $data['cpf'] ?? $admin->cpf;
+
+        if (!empty($data['password']) && !empty($data['new_password'])) {
```

Comment:
Você poderia explicar o porque do campo password, se você já tem o new_password? Me parece desnecessário, primeiro pensei que era pra salvar o new_password somente si o password estivesse condizentes com o do banco, aí eu ia dizer pra você user a rule 'password:api', mas descartei já que você só validou a existência ou não do campo, e o usuário já está autenticado. Não sei se a intenção era validar se ambos são iguais, se for, você poderia usar a regra same. Mas do jeito que está não vi muito sentido. Manda aí uma luz.

Reply:
Certo, o que aconteceu foi o seguinte: eu tava validando de o $data['password'] era igual o que tava no banco com o Hask::check() do laravel. Mesmo eu definindo o atributo password no factory e colocando no model de User para "setar" o password com o Hash::make() a válidação não passava (isso não quer dizer que não estava correto, era algum problema no RefreshDatabase). Como ainda não sabia com seria criptografado o password na autenticação, eu apenas validei a existencia do campo.

O Hash do laravel ele usa por padrão o bcrypt, mas ele um proxy para que possa ser alterado caso necessário a forma de criptografar os dados.

Reply:
@FrankKleiton depois de ver aqui eu vi que tava enviando o usuário errado no test kkk, por isso o password não tava batendo.

Reply:
Sinceramente, não sei porque estava dando errado aí, fiz aqui, e está tudo funcionando perfeitamente:
```
if (!empty($data['password']) && !empty($data['new_password'])) {
     if (Hash::check($data['password'], $admin->password)) {
        $admin->password = $data['new_password'];
    }
}
```
Alterações do teste:
```
$fakeAdmin = factory(User::class)->create(['password' => 'Kaguya', 'user_type' => 'ADMIN_CONAB']);

$dataWithPassword = ['password' => 'Kaguya', 'new_password' => '654321'];
$response = $authenticatedRoute->putJson("/api/conab/admins/$fakeAdmin->id", $dataWithPassword);
$fakeAdmin->refresh();
$this->assertTrue(Hash::check('654321', $fakeAdmin->password));
```
A sugestão que tenho:

```
$data = Validator::make($request->all(), [
            'name' => 'string',
            'email' => 'string|email',
            'cpf' => 'regex:/^[0-9]{3}\.[0-9]{3}\.[0-9]{3}\-[0-9]{2}/',
            'password' => 'string|required_with:new_password',
            'new_password' => 'string',
            'phones.*.number' => 'string|regex:/^\([0-9]{2}\) [0-9]{5}\-[0-9]{4}/'
 ])->validate();

if (!empty($data['new_password'])) {
            if (Hash::check($data['password'], $admin->password)) {
                $admin->password = $data['new_password'];
            }
}
```
Pra esse código essa é minha sugestão, tem algumas coisas que não entendi ainda, por que um admin pode fazer alterações em outro admin? O certo seria o próprio admin fazer suas alterações, não outros. Do jeito que está aí, você vai ter que adicionar a mesma sugestão que fiz do store, a todas as rotas e checar se é o usuário do auth ou o admin com permissão. Eu pessoalmente achava que todas as rotas seriam pro usuário auth e somente o store seria para o adimin com permissão.

Reply:
> @FrankKleiton depois de ver aqui eu vi que tava enviando o usuário errado no test kkk, por isso o password não tava batendo.

Dá próxima vez vou dar refresh na página kk